### PR TITLE
Wire Up Endpoints

### PR DESCRIPTION
**Pull request by meierj-wayscript**
## Change description

> Description here, include any info and attachments relevant to a reviewer

## Checklist

- [x] Application changes have been tested and automated tests have been added, if applicable
- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

## Related issues

> Fixes #xxxx

### CODE REVIEW

File: src/wayscript/settings.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+ROUTES = {
+    "lairs": {"detail": f"{LAIRS_ROUTE}/$id"},
+    "processes": { "detail_expanded": f"{PROCESSES_ROUTE}/$id/detail"},
+    "workspaces": {"detail": f"{WORKSPACES_ROUTE}/$id"},
+}
+
+
+WAYSCRIPT_HOST = os.environ.get("WAYSCRIPT_HOST", "api.wayscript.com")
+WAYSCRIPT_VERSION = os.environ.get("WAYSCRIPT_VERSION", "v0")
```
Comment (meierj-wayscript) [inline src/wayscript/settings.py]: Unnecessary?

File: src/wayscript/settings.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+"""
+import os
+
+
+PROCESSES_ROUTE = "processes"
+LAIRS_ROUTE = "lairs"
+WORKSPACES_ROUTE = "workspaces"
+
+
+ROUTES = {
```
Comment (meierj-wayscript) [inline src/wayscript/settings.py:12]: Very open to any feedback re better ways to ship these routes.

File: src/wayscript/utils.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+
+class WayScriptClient:
+
+    def __init__(self, *args, **kwargs):
+        """Init a wayscript client"""
+        self.session = requests.Session()
+        self.session.headers["authorization"] = get_process_execution_user_token()
+    
+    def _get_url(self, subpath: str, route: str, template_args: dict=None):
```
Comment (meierj-wayscript) [inline src/wayscript/utils.py:28]: Would love to have better words here than "subpath" and "route". It's like "top level route" and "sub route", almost.

File: src/wayscript/context.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    process = get_process()
+    lair_id = process["lair_id"]
+    client = utils.WayScriptClient()
+    response = client.get_lair_detail(lair_id)
+    response.raise_for_status()
+    data = response.json()
+    return data
+
+
+def get_workspace():
```
Comment (jeffbuswell) [inline src/wayscript/context.py:48]: Should we also add the ability to get info about the user executing the call? I think right now we're just defaulting that to the owner of the lair. wdyt?

File: tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+
+@pytest.fixture
+def workspaces_detail_response():
+    """Data from GET /workspaces/<id>"""
+    data = {
+        "file_object": "",
+        "workspace_name": "fbb01af8-c476-4928-8921-b12e9ba91436",
+        "workspace_uuid": WORKSPACE_ID,
+ }
```
Comment (jeffbuswell) [inline tests/conftest.py]: nit: indentation

File: tests/test_context.py [hunk 6]
```diff
@@ -1,4 +1,45 @@
+import pytest
+import responses
+import requests
+
 from wayscript import context as ws_context
+from wayscript import settings as ws_settings
+
+
+def test_always_succeeds():
```
Comment (jeffbuswell) [inline tests/test_context.py]: do we need this?

**Review by jeffbuswell — APPROVED** [review]
Looks awesome overall! Just a couple minor nits - feel free to address in this PR or a subsequent one

File: src/wayscript/context.py (hunk 4 above)
Comment (meierj-wayscript) [reply src/wayscript/context.py:48]: Yeah, can definitely add a method to call out for user detail data. I can add a ticket for it under the context epic.

File: tests/test_context.py (hunk 6 above)
Comment (meierj-wayscript) [reply tests/test_context.py]: Nah

File: src/wayscript/context.py (hunk 4 above)
Comment (meierj-wayscript) [reply src/wayscript/context.py:48]: https://linear.app/wayscript/issue/GEN-246/support-retrieving-execution-user-metadata

**Pull request merged**